lib/setup/setupInstall.js

```diff
@@ -1586,8 +1592,67 @@ function Install(options) {
+                        }
+                        if (ioPack.common.restartAdapters.length && ioPack.common.restartAdapters[0]) {
+                            return new Promise((resolve) => {
+                                tools.getAllInstances(ioPack.common.restartAdapters, objects,instances => {
+                                    if (!instances || !instances.length) {
+                                        if (callback) {
+                                            callback();
+                                            callback = null;
+                                        }
+                                    } else {
+                                        let instancesCount = instances.length;
+                                        for (let r = 0; r < instances.length; r++) {
+                                            objects.getObject(instances[r], (err, obj) => {
+                                                // if instance is enabled
+                                                if (!err && obj && obj.common.enabled) {
+
+                                                    obj.common.enabled = false; // disable instance
+
+                                                    obj.from = `system.host.${tools.getHostName()}.cli`;
+                                                    obj.ts = Date.now();
+
+                                                    objects.setObject(obj._id, obj, err => {
+                                                        if (!err) {
+
+                                                            obj.common.enabled = true; // enable instance
+
+                                                            obj.from = `system.host.${tools.getHostName()}.cli`;
+                                                            obj.ts = Date.now();
+
+                                                            objects.setObject(obj._id, obj, _err => {
+                                                                console.log(`Adapter "${obj._id}" restarted.`);
+                                                                if (!--instancesCount) {
+                                                                    resolve();
+                                                                }
+                                                            });
+                                                        } else {
+                                                            console.error(`Cannot restart adapter "${obj._id}": ${err}`);
+                                                            if (!--instancesCount) {
+                                                                resolve();
+                                                            }
+                                                        }
+                                                    });
+                                                } else if (!--instancesCount) {
```

Comment:
Uh, can we somehow untangle this mess of callback-style and promise-style code?
Like using the async versions of get/setObject and just await in a loop?